target-app/Home/Views/HomeViewController.swift

```diff
@@ -24,22 +45,38 @@ class HomeViewController: UIViewController {
         setupMapConstraints()
     }
     
+    override func viewDidAppear(_ animated: Bool) {
+        LocationManager.shared.requestLocationAuthorization()
+        
+        if CLLocationManager.locationServicesEnabled() {
```

Comment:
This check alone doesn't indicate that we can start listening for location updates. This boolean indicates whether the device has location services enabled in settings.
Instead, you need to check this and also if the user authorized the permissions to the application, which is an async function. Pls see https://developer.apple.com/documentation/corelocation/requesting_authorization_to_use_location_services.

Reply:
Done. Code moved and refactored into `LocationManager` class.